src/cloudsc_pyiface/src/cloudscpytools/cloudsc_data.py

```diff
@@ -0,0 +1,457 @@
+            clsc.expand_mod.expand_r2(lfield, b3field,  nlon, nproma, nlev, ngptot, nblocks)
+            bfield=b3field
+        elif ldims == 4:
+            b4field=np.asfortranarray(np.transpose(np.zeros(shape=dims, dtype="float64")))
+#Note that we are using expand_r3bis here.
+#Original expand_r3 does not pass nlev from python to Fortran
+#            print ("This is 4D field")
+#            print ("lfield")
+#            print (lfield.shape)
+#            print ("bfield")
+#            print (b4field.shape)
+            clsc.expand_mod.expand_r3bis(lfield, b4field,  nlon, nproma, ndim,  ngptot, nblocks)
+            bfield=b4field
+        else:
+            print ("Wrong float ldim")
+    elif lfield.dtype == "bool":
+#Workaround - using type int32, otherwise complains about type disagreement at runtime
+        bfield=np.asfortranarray(np.transpose(np.zeros(shape=dims, dtype='int32')))
+        if ldims == 2:
+            tlfield=lfield.astype('int32')
+            clsc.expand_mod.expand_l1(tlfield, bfield,  nlon, nproma, ngptot, nblocks)
+        else:
+            print ("Wrong bool ldim")
+    elif lfield.dtype == "int32":
+        bfield=np.asfortranarray(np.transpose(np.zeros(shape=dims, dtype='int32')))
+        if ldims == 2:
+            clsc.expand_mod.expand_i1(lfield, bfield,  nlon, nproma, ngptot, nblocks)
+        else:
+            print ("Wrong int ldim")
+    else:
+        print ("Wrong dtype")
+    return bfield
+
+def load_input_fortran_fields(path, nparms, fields):
+    """
+    load_input_fortran_fields returns:
+    - set of variables needed to initiate computation of the Fortran kernel.
+    """
+    nproma =nparms['NPROMA']
+    nlev   =nparms['NLEV']
+    nblocks=nparms['NBLOCKS']
+    argnames_nlev = [
+        'pt', 'pq',
+        'pvfa', 'pvfl', 'pvfi', 'pdyna', 'pdynl', 'pdyni',
+        'phrsw', 'phrlw','pvervel','pap','plu','plude',
+        'psnde', 'pmfu', 'pmfd',
+        'pa', 'psupsat',
+        'plcrit_aer','picrit_aer','pre_ice',
+        'pccn', 'pnice'
+    ]
+    argnames_nlevp = [
+        'paph'
+    ]
+
+    argnames_withnclv= [
+        'pclv','tendency_tmp_cld'
+    ]
+
+    argnames_tend = [
+        'tendency_tmp_t','tendency_tmp_q','tendency_tmp_a'
+    ]
+
+    argnames_scalar = [
+         'kfldx'
+    ]
+
+    argnames_nproma = [
+        'plsm', 'ldcum', 'ktype'
+    ]
+
+
+    with h5py.File(path, 'r') as f:
+        fields['KLON'] = f['KLON'][0]
+        fields['KLEV'] = f['KLEV'][0]
+        fields['PTSPHY'] = f['PTSPHY'][0]
+        klon=fields['KLON']
+
+        for argname in argnames_nlev:
+            print('Loading field:',argname)
+            fields[argname] = field_c_to_fortran((nblocks,nlev,nproma),
+                                                 f[argname.upper()],nparms,klon)
+
+        for argname in argnames_nlevp:
+            print('Loading field:',argname)
+            fields[argname] = field_c_to_fortran((nblocks,nlev+1,nproma),
```

Comment:
If memory serves, hdf5 presents fields as numpy arrays, so there is no need to manage the dimensions explicitly. Instead we could simply interrogate `f[argname].shape` here and manipulate it accordingly, no?

Reply:
Possibly. On the other hand, the explicit list of dimensions annotates the code here. When playing on the two pianos  at  one time, it seems helpful to have more annotations in the code, at least it was for me. Open to your ideas here.